src/transformers/generation_tf_utils.py

```diff
@@ -360,6 +360,7 @@ class TFGenerationMixin:
 
     @property
     def seed_generator(self):
+        warnings.warn("`seed_generator` is deprecated and will be removed in a future version.", UserWarning)
```

Comment:
@sgugger `seed_generator` is public and therefore can't be removed, correct? (despite being recent and exclusively used in `sample`)

Reply:
A deprecation cycle is nicer yes. We can consider removing it in a couple of minors if it was super recent